src/Slot-Core/Slot.class.st

```diff
@@ -1,16 +1,20 @@
 "
 I'm a meta-object for accessing a slot in an Object. 
 
-I define a protocol to read (#read:) and to write (#write:to:) values. 
+I define a reflective protocol to read (#read:) and to write (#write:to:) values. 
+
+	For efficiency, subclasses (e.g. InstanceVariableSlot) override #emitStore: and #emitValue: to not call #read: and/or write:to:
 
 For customizing a subclass can override the meta-object-protocol methods. See subclasses for examples. If your subclass holds state, the following must be overridden:
-- Printing: #definitionString must return the source code string to reconstruct an instance. This will be displayed in the class definition. By default, #definitionString is the printString, so you can also override #printOn: instead.
+- Printing: #printOn: must print the source code string to reconstruct an instance. This will be displayed in the class definition. 
```

Comment:
In the case we do not use #definitionString anymore, I propose to use #storeOn: for this purpose.

Indeed, #printOn: is not expected to print executable code where #storeOn: is expected to do that.

Reply:
Not that easy... the problem is that slots might have additional state that is not part of the definition... e.g. the offset of IndexedSlot and subclasses.

Reply:
I will add it to the backlog... maybe the #printDefinitionOn: idea is not bad after all: this then allows both storeOn: and printOn: to have a good default *and* to be changed indpendently